### ISSUE TITLE
Inconsistent status returned by Get-AzSqlDatabaseImportExportStatus

### ISSUE THREAD

**Issue by aleksandrshvets** [Azure/azure-powershell#13529]
Az.Sql 2.11.1
PowerShell 7.0.3

According to [documentation](https://docs.microsoft.com/en-us/azure/azure-sql/database/database-import?tabs=azure-powershell#using-powershell) the status of database import operation could be verified by checking `Status` field of the object returned by Get-AzSqlDatabaseImportExportStatus cmdlet.
```
$importStatus = Get-AzSqlDatabaseImportExportStatus -OperationStatusLink $importRequest.OperationStatusLink

[Console]::Write("Importing")
while ($importStatus.Status -eq "InProgress") {
    $importStatus = Get-AzSqlDatabaseImportExportStatus -OperationStatusLink $importRequest.OperationStatusLink
    [Console]::Write(".")
    Start-Sleep -s 10
}

[Console]::WriteLine("")
$importStatus
```
However the values of `Status` field can also be: "Pending" and "Running, Progress = 5.00 %". They does not indicate error and the import is considered to be still in the progress.

Can this behavior be fixed so that only "InProgress" `Status` field value will mean that import is still in the progress ?
If it is not possible can someone provide correct code snippet to wait for import operation?

**Comment by dingmeng-xue** [comment]
@aleksandrshvets , thanks for reporting. Could you provide debug message of `Get-AzSqlDatabaseImportExportStatus` execution? You can enable debug message via `$DebugPreference="Continue"`

**Comment by aleksandrshvets** [comment]
@dingmeng-xue 
Sure, here it is:
```
 Get-AzSqlDatabaseImportExportStatus -OperationStatusLink https://management.azure.com/subscriptions/*****/providers/Microsoft.Sql/locations/eastus2/importExportOperationResults/9bdadd1c-2730-4930-876b-da2c2bb88906-a***?api-version=2020-02-02-preview
DEBUG: Sought all Az modules and got latest version 3.6.1
DEBUG: 4:44:08 PM - GetAzureSqlDatabaseImportExportStatus begin processing with ParameterSet '__AllParameterSets'.
DEBUG: 4:44:08 PM - using account id 'a*****.com'...
DEBUG: [Common.Authentication]: Authenticating using Account: 'a*****.com', environment: 'AzureCloud', tenant: '****'
DEBUG: SharedTokenCacheCredential.GetToken invoked. Scopes: [ https://management.core.windows.net//.default ] ParentRequestId:
DEBUG: SharedTokenCacheCredential.GetToken succeeded. Scopes: [ https://management.core.windows.net//.default ] ParentRequestId:  ExpiresOn: 2020-11-19T14:39:31.0000000+00:00
DEBUG: SharedTokenCacheCredential.GetToken invoked. Scopes: [ https://management.core.windows.net//.default ] ParentRequestId:
DEBUG: SharedTokenCacheCredential.GetToken succeeded. Scopes: [ https://management.core.windows.net//.default ] ParentRequestId:  ExpiresOn: 2020-11-19T14:39:31.0000000+00:00
DEBUG: [Common.Authentication]: Received token with LoginType 'User', Tenant: '****', UserId: 'a*****.com'

OperationStatusLink          : https://management.azure.com/subscriptions/*****/providers/Microsoft.Sql/locations/eastus2/importExportOperationResults/9bdadd1c-2730-4930-876b-da2
                               c2bb88906-****-***?api-version=2020-02-02-preview
ErrorMessage                 :
LastModifiedTime             : 11/19/2020 1:43:05 PM
QueuedTime                   : 11/19/2020 1:43:02 PM
RequestType                  : Import
Status                       : Running, Progress = 5.00 %
StatusMessage                :
PrivateEndpointRequestStatus : {}

DEBUG: AzureQoSEvent: CommandName - Get-AzSqlDatabaseImportExportStatus; IsSuccess - True; Duration - 00:00:04.2681663;
DEBUG: Finish sending metric.
DEBUG: 4:44:13 PM - GetAzureSqlDatabaseImportExportStatus end processing.
```

Just to clarify. The problem is not that import fails. The problem is that there is no documented way to wait until import is done.
`Status` field can have a set values which are indicates that import is still being in progress. And those values are not documented.

**Comment by martingbrown** [comment]
I've been getting this on my Azure DevOps deployment, but have been unable to reproduce it on my local machine. It seems that the versions of Az.Sql are different 2.10.0 on Azure DevOps and 2.1.2 locally. I decided to do a bit of digging in the code and I think this may be a regression that was introduced with commit 56dbaa9e. 

`ImportExportDatabaseAdapter.GetStatus(string)` was switched from calling `ImportExportOperationStatusResponse Communicator.GetStatus(string)` to `ImportExportOperationResult Communicator.GetOperationStatus(string)` which I suspect is giving different results. I also notice that this method is no longer setting StatusMessage where as it was before this commit.

**Comment by sredman** [comment]
This issue has been fixed in the latest version of Azure Powershell v5.3.0 (see attached PR). However, while I was working on this issue, I noticed that the status values being returned by the SQL API were wrong. That fix has been coded and checked in but it will be another couple of weeks before it can be deployed. For now, I recommend using v4.8.0 of Powershell which is not affected by either issue.
